File: C++ Implementation/Algorithms/Sorting/InsertionSort.cpp
void insertionSort(int arr[], int n){
    for(int i=0; i<n; i++){
        bool swap = false;

        for(int j=i-1; j>=0; j--){
            // j+1 = current element
            // j = previous element
            if(arr[j+1] < arr[j]){
                // swap
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                swap = true;
            }

            if(!swap){
                break;
            }
        }
    }
}

int main(){
    int n = 5;
    int arr[5] = {5, 1, 4, 2, 8};
    insertionSort(arr, n);
    return 0;    
}

File: Python Implementation/Algorithms/Sorting/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        current_index = i
        # j pointer starts at i - 1
        j = i - 1

        while j >= 0 and arr[j+1] < arr[j]:
            temp = arr[j]
            arr[j] = arr[j+1]
            arr[j+1] = temp
            j -= 1
        print(arr)
    



    return arr

if __name__ == "__main__":
    x = [5, 2, 9,3, 1, 6]
    print(insertion_sort(x))